dataset/utils.py

```diff
@@ -15,6 +15,5 @@ def traverse_clf_dataset(dataset_path: str):
         _, _, dir_files = next(walk(base_dir_path))
         dir_files_paths = map(lambda file_: join(base_dir_path, file_), dir_files)
         dir_files_paths = filter(_in_c_family, dir_files_paths)
-        for file in dir_files_paths:
-            files.append(file)
+        files += [file for file in dir_files_paths]
```

Comment:
Maybe it's better to write
```
files += list(dir_files_paths)
```
So it wouldn't be confusing with this dumb list comprehension.

Reply:
Also, you can implement this function as a generator:
```
def traverse_clf_dataset(dataset_path: str):
    _, base_dirs, _ = next(walk(dataset_path))
    base_dirs_paths = map(lambda file_: join(dataset_path, file_), base_dirs)
    for base_dir_path in base_dirs_paths:
         _, _, dir_files = next(walk(base_dir_path))
         dir_files_paths = map(lambda file_: join(base_dir_path, file_), dir_files)
         yield filter(_in_c_family, dir_files_paths)
```